onnx/checker.cc

```diff
@@ -424,6 +419,12 @@ void check_sparse_tensor(
 
   const TensorProto& values = sparse_tensor_proto.values();
   check_tensor(values, ctx);
+  // SparseTensors make use of values name
+  // and this is important for use as inputs or sparse initializers
+  enforce_has_field(values, name);
+  if (values.name().empty()) {
```

Comment:
One question: TensorProtos and SparseTensorProtos can appear in two contexts. (a) In initializers, and (b) As an attribute-value in a node. Is the name compulsorily needed in both contexts or only when they are used in initializers? This is more a question about how to interpret the existing standard/spec. My interpretation is that the name is required in case (a), not necessarily in case (b). If so, I think we can have this check when checking initializers, instead of at this place. In either case, I think we should have the same requirement/check for dense initializers as well as sparse initializers.

Reply:
I agree that the spec is not very explicit about this. I am inferring it indirectly, since onnx.proto specifies "field MUST be present" if it is compulsory for all instances. Further, AttributeProto has its own name.

Reply:
If there are no other opinions, I will make the name optional for sparse tensor but enforce it for sparse initializers.

Reply:
I think that is best. I also inferred the spec the similar way. It seems like the name is required for initializers and not in the case of attribute value